### DISCUSSION TITLE
Yarn vs NPM

### DISCUSSION THREAD

**Discussion by silviogutierrez** [silviogutierrez/reactivated#138] [RFC]
Projects created using Reactivated use Yarn Classic. When Yarn first came out, it was fast. _dramatically_ faster than NPM. It also encouraged a lock file for deterministic builds. Finally, it had support for workspaces built-in. All while being fully compatible with the NPM ecosystem.

These days, NPM has caught up in speed. It also supports lockfiles and workspaces. More importantly, Yarn 2+ no longer aims to be a drop-in replacement for NPM.

The point of this RFC isn't to discuss the merit of Yarn 2+ features, but whether it makes sense to keep Yarn Classic as the package manager.

## Pros

-   It's there. It works. It's fast.
-   Running scripts doesn't require `yarn run <command>`. You can just type `yarn <command>`.

## Cons

-   It's an additional dependency.
-   Development on the "Classic" Yarn is basically done.

**Comment by stlk** [comment]
I definitely recommend moving to npm. It's completely fine now. Last thing missing was `npx` command. Just managed to start https://github.com/silviogutierrez/reactivated/tree/main/development without nix and yarn. Only thing I had to do was replace `yarn` with `npx` in https://github.com/silviogutierrez/reactivated/blob/eba2be26d204fbd4fb4731161e4194cf15e34636/reactivated/processes.py#L11

**Reply by silviogutierrez** [reply]
I think the replacement would be `npm run tsc` right? `npx` will go and install it if it's not installed, which is not quite what we want.

**Reply by silviogutierrez** [reply]
But point taken about moving to `npm` over `yarn`

**Reply by stlk** [reply]
> I think the replacement would be `npm run tsc` right? `npx` will go and install it if it's not installed, which is not quite what we want.

Just tried it to be sure. `npm run tsc` doesn't work. :)

- `npm run` is only for running scripts defined in `package.json`.
- `npx` will first look for local package and will pull it from the registry only if it's not available.

**Reply by silviogutierrez** [reply]
Oh wow, you're right. I didn't realize `yarn tsc` was *not* the same as `npm run tsc`.

Really, probably best to just do `node_modules/.bin/tsc` as I do elsewhere. Much faster.

**Comment by silviogutierrez** [comment]
Resolved by #199
